### PR TITLE
new features for android notifications

### PR DESCRIPTION
**Pull request by npomfret**
* fixed 'id' and 'number' field parsing - docs specified (correctly) that these fields are numeric but the code was parsing them as strings which would return null
* added log warnings for when user uses strings instead of ints for 'id' and 'number' fields
* surrounded message parsing with a try/catch block to prevent malformed messages from crashing the app
* added support for 'tag' field
* added support for 'group' field
* added support for 'playSound' field
* added support for custom vibration ('vibrate' and 'vibration' field)
* updated docs to reflect new functionality

### CODE REVIEW

**Comment by fermuch** [comment]
(I'm not a java developer, so I'm not sure if what I'm saying is a problem with my setup or with this PR)

After testing your PR, I get the following error:

```
* What went wrong:
A problem occurred configuring project ':app'.
> A problem occurred configuring project ':react-native-push-notification'.
   > failed to find Build Tools revision 23.0.3
```

Changing the version to `23.0.0` fixed the issue, and everything seems to work alright.

**Comment by npomfret** [comment]
Thanks for testing.  Not sure what that means, but cheers for fixing

File: android/build.gradle [hunk 1]
```diff
@@ -2,7 +2,7 @@ apply plugin: 'com.android.library'
 
 android {
     compileSdkVersion 23
-    buildToolsVersion "23.0.1"
+    buildToolsVersion "23.0.3"
```
Comment (fermuch) [inline android/build.gradle]: This is the change that broke my build. Reverting to 23.0.1 fixed the problem.

**Comment by npomfret** [comment]
https://github.com/zo0r/react-native-push-notification/pull/102

**Pull request closed without merging**